### PR TITLE
Reserve to ListReserve in intrinsic_function_registry.h

### PR DESCRIPTION
**Pull request by rajeshkotaru**

### CODE REVIEW

File: src/libasr/pass/intrinsic_function_registry.h [hunk 1]
```diff
@@ -518,9 +518,9 @@ namespace IntrinsicScalarFunctionRegistry {
                 {"minexponent", {&MinExponent::create_MinExponent, &MinExponent::eval_MinExponent}},
                 {"maxexponent", {&MaxExponent::create_MaxExponent, &MaxExponent::eval_MaxExponent}},
                 {"list.index", {&ListIndex::create_ListIndex, &ListIndex::eval_list_index}},
-                {"list.reverse", {&ListReverse::create_ListReverse, &ListReverse::eval_ListReverse}},
+                {"list.reverse", {&ListReverse::create_ListReverse, &ListReverse::eval_list_reverse}},
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/intrinsic_function_registry.h:521]:
```suggestion
                {"list.reverse", {&ListReverse::create_ListReverse, &ListReverse::eval_ListReverse}},
```

File: src/libasr/pass/intrinsic_function_registry.h [hunk 2]
```diff
@@ -518,9 +518,9 @@ namespace IntrinsicScalarFunctionRegistry {
...
                 {"list.pop", {&ListPop::create_ListPop, &ListPop::eval_list_pop}},
-                {"reserve", {&Reserve::create_Reserve, &Reserve::eval_Reserve}},
+                {"list.reserve", {&ListReserve::create_ListReserve, &ListReserve::eval_list_reserve}},
```
Comment (Shaikh-Ubaid) [inline src/libasr/pass/intrinsic_function_registry.h:523]:
```suggestion
                {"list.reserve", {&ListReserve::create_ListReserve, &ListReserve::eval_ListReserve}},
```

**Comment by Shaikh-Ubaid** [comment]
I think you also need to update at the following place:

https://github.com/lfortran/lfortran/blob/7f05a7ad01842c72c7d505efaf1dfa3e7435c2a1/src/libasr/intrinsic_func_registry_util_gen.py#L89

**Comment by rajeshkotaru** [comment]
thanks for suggestions

**Comment by rajeshkotaru** [comment]
i have made the changes u said but still checks were not successful

**Comment by Shaikh-Ubaid** [comment]
From the CI, it seems the project does not build. Which means there are still some places where the renaming needs to be updated/propagated. Try looking into the specific lines where the error occurs and see if you can figure out where other places renaming needs to be updated.

**Comment by rajeshkotaru** [comment]
yes, i will check for other places.

**Pull request closed without merging**